File: api_node/src/models/afectacionTipo.js
module.exports = (sequelize, DataTypes) => {
  return sequelize.define('AfectacionTipo', {
    id: { type: DataTypes.CHAR(2), primaryKey: true },
    descripcion: { type: DataTypes.STRING(50), allowNull: false },
    letra: { type: DataTypes.CHAR(1), allowNull: false },
    codigo: { type: DataTypes.CHAR(4), allowNull: false },
    porcentaje: { type: DataTypes.DECIMAL(4, 2), allowNull: false }
  }, {
    tableName: 'afectacion_tipos',
    timestamps: false
  });
};

File: demo/src/main/java/com/example/ventas/demo/model/entity/AfectacionTipo.java
package com.example.ventas.demo.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "afectacion_tipos")
public class AfectacionTipo {
    @Id
    @Column(length = 2)
    private String id;

    @Column(length = 50, nullable = false)
    private String descripcion;

    @Column(length = 1, nullable = false)
    private String letra;

    @Column(length = 4, nullable = false)
    private String codigo;

    @Column(precision = 4, scale = 2, nullable = false)
    private BigDecimal porcentaje;
}
